### PR TITLE
M3-3140 Add: Ability to update username from profile -> display

### PR DESCRIPTION
**Pull request by johnwcallahan**
## Description

This makes the "Username" field on Profile -> Display editable.

Same difficulty present in https://github.com/linode/manager/pull/4931 exists here, namely: 

a) Username is only editable at `/account`
b) Email is only editable at `/profile`

This means we need separate submit buttons for each.

- I created a new component called `SingleTextFieldForm`, since this is a pattern we end up using quite a bit (I used it twice on this page).
- Delted EmailChangeForm.tsx in favor of putting everything in DisplaySettings.tsx.
- Removed the tests for EmailChangeForm.tsx and DisplaySettings.tsx in favor of new tests for SingleTextFieldForm.

## Note to Reviewers

When testing, please check account admins, restricted users, etc. Restricted users should see:

Screen Shot 2020-10-29 at 3 43 35 PM

### CODE REVIEW

File: packages/manager/src/features/Profile/DisplaySettings/DisplaySettings.tsx [hunk 1]
```diff
@@ -1,130 +1,126 @@
...
+  const updateTimezone = (newTimezone: string) => {
+    setUsernameResetToken(v4());
+    setEmailResetToken(v4());
+    return updateProfile({ timezone: newTimezone });
+  };
 
-const connected = connect(mapStateToProps, mapDispatchToProps);
+  return (
+    <>
+      <Paper className={classes.paper}>
```
Comment (Jskobos) [inline packages/manager/src/features/Profile/DisplaySettings/DisplaySettings.tsx:75]: What do you think about setting a fixed height for these? The jumping around on success/error is imo worse than a bit of wasted space in the default state.

**Review by Jskobos — APPROVED** [review]
Looks good. Only possible objection is: the new component is uncontrolled, and in my experience that only works until it doesn't and we end up pushing down a `value` prop. You're using keys here, which works well, but if you needed to actively update the other forms to reflect something here we'd be out of luck.

**Comment by johnwcallahan** [comment]
> Looks good. Only possible objection is: the new component is uncontrolled, and in my experience that only works until it doesn't and we end up pushing down a `value` prop. You're using keys here, which works well, but if you needed to actively update the other forms to reflect something here we'd be out of luck.

Fair enough. No matter how we do it, it's strange to have multiple "mini forms" on one page like this. The idea was that the new component is a form by itself that handles its own state; I was trying to avoid this (first pass) where everything was duplicated:

https://github.com/linode/manager/blob/79442de314c9fadb93dc697ca627392ab2a91fe8/packages/manager/src/features/Profile/DisplaySettings/EmailChangeForm.tsx#L49

Not saying I love it though, as I don't love anything about this page 😆

**Review by WilkinsKa1 — APPROVED** [review]
Working as expected for me on regular (admin) and restricted account levels 👍

File: packages/manager/src/features/Profile/DisplaySettings/DisplaySettings.tsx (hunk 1 above)
Comment (johnwcallahan) [reply packages/manager/src/features/Profile/DisplaySettings/DisplaySettings.tsx:75]: I tried this; seemed a little unsettling to me. We have this problem throughout the app; I'm not sure of a good solution without re-working the UX pattern for success messages 🤷

**Pull request merged**